test/e2e/network_segmentation.go

```diff
@@ -1820,7 +1926,7 @@ var nadToUdnParams = map[string]string{
 
 func generateUserDefinedNetworkManifest(params *networkAttachmentConfigParams) string {
```

Comment:
I think that since you're adding a lot of optional attributes you could walk the extra mile of turning this into a builder.

Just an opinion, not a must.

Reply:
OK. In a perfect world, that'd be OK, since we could push a PR on top of this to refactor this code.